manta-crypto/src/merkle_tree/inner_tree.rs

```diff
@@ -768,11 +822,25 @@ where
     /// Returns the path at `leaf_index` without checking if `leaf_index` is later than the
     /// starting index of this tree.
     #[inline]
-    pub fn path_unchecked(&self, leaf_index: Node) -> InnerPath<C>
+    pub fn path_unchecked(&self, leaf_index: Node) -> Option<InnerPath<C>>
     where
         InnerDigest<C>: Clone,
     {
-        self.inner_tree.path(leaf_index)
+        Some(InnerPath::new(
+            leaf_index,
+            (1..C::HEIGHT)
+                .map(|level| {
+                    self.get(
+                        InnerNode::new(0, leaf_index)
+                            .ancestor(level)
+                            .expect("Querying the ancestor does not fail within height bounds."),
+                    )
+                })
+                .collect::<Option<Vec<&InnerDigest<C>>>>()?
+                .iter()
+                .map(|&x| x.clone())
+                .collect(),
+        ))
```

Comment:
You can either clone before or after the first collection, but if you want to do it after then you should do the following:

```
.collect::<Option<Vec<_>>>()?
.into_iter()
.cloned()
.collect()
```